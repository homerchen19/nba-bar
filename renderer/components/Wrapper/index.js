import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Router from 'next/router';
import { TabBar, Flex } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '../../styles/theme';
import calendarWhiteIcon from '../../../resources/calendar_white.svg';
import calendarGreyIcon from '../../../resources/calendar_grey.svg';
import trophyWhiteIcon from '../../../resources/trophy_white.svg';
import trophyBlackIcon from '../../../resources/trophy_grey.svg';
import settingsWhiteIcon from '../../../resources/settings_white.svg';
import settingsBlackIcon from '../../../resources/settings_grey.svg';

const Layout = styled.div`
  position: fixed;
  top: 0;
  width: 100vw;
  height: 100vh;
`;

const Content = styled(Flex)`
  flex: 1;
  width: 100%;
  height: 100%;
  padding: 0;
`;

const Icon = styled.div`
  width: 22px;
  height: 22px;
  background: url(${props => props.url}) center center / 21px 21px no-repeat;
  cursor: pointer;
`;

const Title = styled.span`
  cursor: pointer;
`;

const Wrapper = ({ currentTab, children }) => (
  <Layout>
    <TabBar
      tintColor={colors.white}
      barTintColor={colors.blue}
      unselectedTintColor={colors.liteGrey}
    >
      <TabBar.Item
        key="Schedule"
        title={<Title>Schedule</Title>}
        selected={currentTab === 1}
        icon={<Icon url={calendarGreyIcon} />}
        selectedIcon={<Icon url={calendarWhiteIcon} />}
        onPress={() => {
          Router.push(`/home`);
        }}
      >
        <Content direction="column">{currentTab === 1 && children}</Content>
      </TabBar.Item>
      <TabBar.Item
        key="Standing"
        title={<Title>Standing</Title>}
        selected={currentTab === 2}
        icon={<Icon url={trophyBlackIcon} />}
        selectedIcon={<Icon url={trophyWhiteIcon} />}
        onPress={() => {
          Router.push(`/standings`);
        }}
      >
        <Content direction="column">{currentTab === 2 && children}</Content>
      </TabBar.Item>
      <TabBar.Item
        key="Settings"
        title={<Title>Settings</Title>}
        selected={currentTab === 3}
        icon={<Icon url={settingsBlackIcon} />}
        selectedIcon={<Icon url={settingsWhiteIcon} />}
        onPress={() => {
          Router.push(`/settings`);
        }}
      >
        <Content direction="column">{currentTab === 3 && children}</Content>
      </TabBar.Item>
    </TabBar>
  </Layout>
);

Wrapper.propTypes = {
  currentTab: PropTypes.number.isRequired,
  children: PropTypes.element,
};

Wrapper.defaultProps = {
  children: <Fragment />,
};

export default Wrapper;
