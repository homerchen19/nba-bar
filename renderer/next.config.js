const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
