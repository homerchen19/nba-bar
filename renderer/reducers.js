import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
