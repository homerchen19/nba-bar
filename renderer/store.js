import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
