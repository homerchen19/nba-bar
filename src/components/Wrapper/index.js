import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { TabBar, Flex } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '../../styles/theme';
import calendarWhiteIcon from '../../../resources/calendar_white.svg';
import calendarGreyIcon from '../../../resources/calendar_grey.svg';
import trophyWhiteIcon from '../../../resources/trophy_white.svg';
import trophyBlackIcon from '../../../resources/trophy_grey.svg';
import settingsWhiteIcon from '../../../resources/settings_white.svg';
import settingsBlackIcon from '../../../resources/settings_grey.svg';

const Layout = styled.div`
  position: fixed;
  top: 0;
  width: 100%;
  height: 100%;
`;

const Content = styled(Flex)`
  width: 100%;
  height: 100%;
  padding: 0;
  flex: 1;
`;

const Icon = styled.div`
  width: 22px;
  height: 22px;
  background: url(${props => props.url}) center center / 21px 21px no-repeat;
  cursor: pointer;
`;

const Title = styled.span`
  cursor: pointer;
`;

const Wrapper = ({ currentTab, history, children, standings, settings }) => (
  <Layout>
    <TabBar
      tintColor={colors.white}
      barTintColor={colors.blue}
      unselectedTintColor={colors.liteGrey}
    >
      <TabBar.Item
        key="Schedule"
        title={<Title>Schedule</Title>}
        selected={currentTab === 1}
        icon={<Icon url={calendarGreyIcon} />}
        selectedIcon={<Icon url={calendarWhiteIcon} />}
        onPress={() => {
          history.push(`/`);
        }}
      >
        <Content direction="column">{children}</Content>
      </TabBar.Item>
      <TabBar.Item
        key="Standing"
        title={<Title>Standing</Title>}
        selected={currentTab === 2}
        icon={<Icon url={trophyBlackIcon} />}
        selectedIcon={<Icon url={trophyWhiteIcon} />}
        onPress={() => {
          history.push(`/standings`);
        }}
      >
        <Content direction="column">{standings}</Content>
      </TabBar.Item>
      <TabBar.Item
        key="Settings"
        title={<Title>Settings</Title>}
        selected={currentTab === 3}
        icon={<Icon url={settingsBlackIcon} />}
        selectedIcon={<Icon url={settingsWhiteIcon} />}
        onPress={() => {
          history.push(`/settings`);
        }}
      >
        <Content direction="column">{settings}</Content>
      </TabBar.Item>
    </TabBar>
  </Layout>
);

Wrapper.propTypes = {
  currentTab: PropTypes.number.isRequired,
  history: PropTypes.object.isRequired,
  children: PropTypes.element,
  settings: PropTypes.element,
  standings: PropTypes.element,
};

Wrapper.defaultProps = {
  children: <Fragment />,
  settings: <Fragment />,
  standings: <Fragment />,
};

export default Wrapper;
