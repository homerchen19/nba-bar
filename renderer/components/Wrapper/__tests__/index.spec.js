import React from 'react';
import renderer from 'react-test-renderer';

import Wrapper from '..';

it('be defined', () => {
  expect(Wrapper).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    currentTab: 1,
    history: {
      push: jest.fn(),
    },
  };
  const tree = renderer.create(<Wrapper {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
