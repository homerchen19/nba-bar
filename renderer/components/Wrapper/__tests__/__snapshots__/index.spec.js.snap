// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="Layout-sc-1qlvb6x-0 iYCmAx"
>
  <div
    className="am-tab-bar"
  >
    <div
      className="am-tabs am-tabs-horizontal am-tabs-bottom"
    >
      <div
        className="am-tabs-content-wrap"
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "left": "0%",
            "position": "relative",
            "touchAction": "pan-x pan-y",
          }
        }
      >
        <div
          className="am-tabs-pane-wrap am-tabs-pane-wrap-active"
          style={Object {}}
        >
          <div
            className="am-tab-bar-item"
          >
            <div
              className="am-flexbox Wrapper__Content-sc-5wpvpv-0 bGWeRy am-flexbox-dir-column am-flexbox-align-center"
            />
          </div>
        </div>
        <div
          className="am-tabs-pane-wrap am-tabs-pane-wrap-inactive"
          style={Object {}}
        >
          <div
            className="am-tab-bar-item"
          >
            <div
              className="am-flexbox Wrapper__Content-sc-5wpvpv-0 bGWeRy am-flexbox-dir-column am-flexbox-align-center"
            />
          </div>
        </div>
        <div
          className="am-tabs-pane-wrap am-tabs-pane-wrap-inactive"
          style={Object {}}
        />
      </div>
      <div
        className="am-tabs-tab-bar-wrap"
      >
        <div
          className="am-tab-bar-bar"
          style={
            Object {
              "backgroundColor": "#0069b9",
            }
          }
        >
          <div
            className="am-tab-bar-tab"
            onClick={[Function]}
          >
            <div
              className="am-tab-bar-tab-icon"
              style={
                Object {
                  "color": "#f7f7f7",
                }
              }
            >
              <div
                className="Wrapper__Icon-sc-5wpvpv-1 bcfahj"
              />
            </div>
            <p
              className="am-tab-bar-tab-title"
              style={
                Object {
                  "color": "#f7f7f7",
                }
              }
            >
              <span
                className="Wrapper__Title-sc-5wpvpv-2 cCyuiF"
              >
                Schedule
              </span>
            </p>
          </div>
          <div
            className="am-tab-bar-tab"
            onClick={[Function]}
          >
            <div
              className="am-tab-bar-tab-icon"
              style={
                Object {
                  "color": "#8D99AE",
                }
              }
            >
              <div
                className="Wrapper__Icon-sc-5wpvpv-1 bcfahj"
              />
            </div>
            <p
              className="am-tab-bar-tab-title"
              style={
                Object {
                  "color": "#8D99AE",
                }
              }
            >
              <span
                className="Wrapper__Title-sc-5wpvpv-2 cCyuiF"
              >
                Standing
              </span>
            </p>
          </div>
          <div
            className="am-tab-bar-tab"
            onClick={[Function]}
          >
            <div
              className="am-tab-bar-tab-icon"
              style={
                Object {
                  "color": "#8D99AE",
                }
              }
            >
              <div
                className="Wrapper__Icon-sc-5wpvpv-1 bcfahj"
              />
            </div>
            <p
              className="am-tab-bar-tab-title"
              style={
                Object {
                  "color": "#8D99AE",
                }
              }
            >
              <span
                className="Wrapper__Title-sc-5wpvpv-2 cCyuiF"
              >
                Settings
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
